src/libstd/sys/windows/path.rs

```diff
@@ -82,11 +82,33 @@ pub fn parse_prefix(path: &OsStr) -> Option<Prefix<'_>> {
         return None;
     }
 
-    fn parse_two_comps(mut path: &[u8], f: fn(u8) -> bool) -> Option<(&[u8], &[u8])> {
-        let first = &path[..path.iter().position(|x| f(*x))?];
-        path = &path[(first.len() + 1)..];
-        let idx = path.iter().position(|x| f(*x));
-        let second = &path[..idx.unwrap_or(path.len())];
+    /// Split path component that matches predicate `f`.
+    ///
+    /// Returns [`None`] if there is no separator in path.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::::{split_path_component, is_verbatim_sep};
+    /// assert_eq!(
+    ///     split_path_component(br"server\share", is_verbatim_sep),
+    ///     Some((&b"server"[..], &b"share"[..])),
+    /// );
+    /// assert_eq!(
+    ///     split_path_component(br"server\", is_verbatim_sep),
+    ///     Some((&b"server"[..], &b""[..]))
+    /// );
+    /// ```
+    pub fn split_path_component<'a>(
+        path: &'a [u8],
+        f: fn(u8) -> bool,
+    ) -> Option<(&'a [u8], &'a [u8])> {
+        let idx = path.iter().position(|&x| f(x))?;
+        // Remove the path component separator
+        // SAFETY: `s[idx + 1..] == s[s.len()..]`
+        let (first, path) = unsafe { (path.get_unchecked(..idx), path.get_unchecked(idx + 1..)) };
+        let idx = path.iter().position(|&x| f(x)).unwrap_or(path.len());
+        let second = unsafe { path.get_unchecked(..idx) };
```

Comment:
Why does this check for a second separator and then ignores the tail? The old function does that, yes. But if that's indeed required by the callers, then this fact should be in the docs of this new function. *And* the function should probably called differently; a name that emphasizes that "the first two components are returned".